packages/ui/components/ui/tx/actions-views/ibc-relay.tsx

```diff
@@ -4,93 +4,119 @@ import {
 import { uint8ArrayToBase64 } from '@penumbra-zone/types/base64';
+import { ReactElement } from 'react';
+import { Packet } from '@penumbra-zone/protobuf/ibc/core/channel/v1/channel_pb';
 import { getUtcTime } from './isc20-withdrawal.tsx';
-import { useMemo } from 'react';
 
-// Packet data stored as json string encoded into bytes
-const parsePacket = ({ packet }: MsgRecvPacket): FungibleTokenPacketData | undefined => {
-  if (!packet?.data) {
+// Attempt to parse data w/ fallbacks if unknown or failures
+const ParsedPacketData = ({
+  data,
+  dataParser,
+}: {
+  data: Uint8Array;
+  dataParser?: (arg: Uint8Array) => ReactElement;
+}) => {
+  if (!data.length) {
     return undefined;
   }
 
+  if (!dataParser) {
+    return <ActionDetails.Row label='Data'>Unknown packet data</ActionDetails.Row>;
+  }
+
   try {
-    const dataString = new TextDecoder().decode(packet.data);
-    return FungibleTokenPacketData.fromJsonString(dataString);
+    return dataParser(data);
   } catch (e) {
-    return undefined;
+    return <ActionDetails.Row label='Data'>Error while parsing data</ActionDetails.Row>;
   }
 };
 
-const MsgResvComponent = ({ packet }: { packet: MsgRecvPacket }) => {
-  const packetData = useMemo(() => parsePacket(packet), [packet]);
+const PacketRows = ({
+  packet,
+  dataParser,
+}: {
+  packet: Packet;
+  dataParser?: (arg: Uint8Array) => ReactElement;
+}) => {
+  return (
+    <>
+      <ParsedPacketData data={packet.data} dataParser={dataParser} />
+      {!!packet.sequence && (
+        <ActionDetails.Row label='Sequence'>{Number(packet.sequence)}</ActionDetails.Row>
+      )}
+      {!!packet.sourcePort && (
+        <ActionDetails.Row label='Source Port'>{packet.sourcePort}</ActionDetails.Row>
+      )}
+      {!!packet.sourceChannel && (
+        <ActionDetails.Row label='Source Channel'>{packet.sourceChannel}</ActionDetails.Row>
+      )}
+      {!!packet.destinationPort && (
+        <ActionDetails.Row label='Destination Port'>{packet.destinationPort}</ActionDetails.Row>
+      )}
+      {!!packet.destinationChannel && (
+        <ActionDetails.Row label='Destination Channel'>
+          {packet.destinationChannel}
+        </ActionDetails.Row>
+      )}
+      {!!packet.timeoutHeight?.revisionHeight && (
+        <ActionDetails.Row label='Timeout revision height'>
+          {Number(packet.timeoutHeight.revisionHeight)}
+        </ActionDetails.Row>
+      )}
+      {!!packet.timeoutHeight?.revisionNumber && (
+        <ActionDetails.Row label='Timeout revision number'>
+          {Number(packet.timeoutHeight.revisionNumber)}
+        </ActionDetails.Row>
+      )}
+      {!!packet.timeoutTimestamp && (
+        <ActionDetails.Row label='Timeout timestamp'>
+          {getUtcTime(packet.timeoutTimestamp)}
+        </ActionDetails.Row>
+      )}
+    </>
+  );
+};
+
+// Packet data stored as json string encoded into bytes
+const parseRecvPacket = (packetData: Uint8Array) => {
```

Comment:
Was informed that all of these messages expect to have a `FungibleTokenPacketData` inside. Will refactor a bit for that case.